### ISSUE TITLE
Playlist cannot be added

### ISSUE THREAD

**Issue by rafaeldamasceno** [TripleFrequency/AthamePlugin.Deezer#21]
Playlist is https://www.deezer.com/us/playlist/3866715346

Here's the error: https://i.imgur.com/a4HONCG.png

**Comment by rafaeldamasceno** [comment]
Also, when this happens it leaves Athame in an unusable state, as I can't start a download. There's a message saying that there might be something wrong with my login.

**Comment by TripleFrequency** [comment]
Intereseting. When I attempted to add the playlist, I had no issues. Could you re-attempt this direct after starting Athame? "not logged in" is typically related to an expired session. Also, could you post your Athame logs here? Or take another screenshot if it happens again?

**Comment by rafaeldamasceno** [comment]
It was right after starting it! So, it did happen again and there was a log, but it seems it wasn't from this exact occasion, as the timestamp was wrong. However, when trying to replicate the issue, no other log was produced :( All I get is the exact same error message.

**Comment by TripleFrequency** [comment]
What version of the plugin are you using?

**Comment by rafaeldamasceno** [comment]
I was using 1.2.7-RC3, which was the latest at the time of writing. However, I just got the same behavior with 1.3.0-RC1.

**Comment by TripleFrequency** [comment]
Does it occur immediately, or does it take a while (well into loading the playlist)?

**Comment by rafaeldamasceno** [comment]
It was around a minute before the error message was displayed. Here's the message again: https://i.imgur.com/9caCLW0.png

Also, nevermind what I said about the logs before. I went through the trash and found them there, but they were errors that you fixed in the latest version, and unrelated to this one.

**Comment by TripleFrequency** [comment]
I really can't do anything about this without more information. I can't reproduce it, which means I can't fix it. If you want to debug it, you could do it a few ways, but i'll explain the easiest.

Download something like Fiddler or Wireshark (I recommend Fiddler for ease of use), start recording your internet traffic, add the playlist, when it errors, find the latest link request inside the application that calls to deezer.com/ajax/gw-light.

Then copy the contents of the request (such as the response body). Let me know if you need further information to do this.

**Comment by rafaeldamasceno** [comment]
Hey, it seemed that Athame made two requests at a time, and here's the last couple API answers: https://pastebin.com/GadQh8aQ

After this, it simply presented that message. If you need some other log or the full HTTP response, do let me know.

**Comment by TripleFrequency** [comment]
I would also like to see the response of the last request in text form. So I can see if the information you are receiving is invalid, or something of the sort.

**Comment by rafaeldamasceno** [comment]
So, here's the full response and yes, it did crash at this exact track twice: https://pastebin.com/iu25bhBS

**Comment by bascurtiz** [comment]
Tried the playlist on TS: all good here.
Using Win10x64 + Athame 0.4.5 + AthamePlugin.Deezer 1.3 RC1

The last 9000+ tracks I d/led spread over 76 playlists, infact, went good.
Using either 1.2.7 RC3, RC2 or 1.3 RC1.

**Comment by TripleFrequency** [comment]
Also, the JSON response you gave me seems perfectly valid. I simply can't understand what could be going wrong here. I'll keep investigating more, though. If you come across any more telling information, please post about it.

Also, thanks for the assist, @bascurtiz.

**Comment by rafaeldamasceno** [comment]
So, I got the exact same error on another machine, using the latest versions of Athame and AthamePlugin.Deezer. They differ on the Windows 10 Pro x64 build. One has the Fall Creators Update, the other just has the Creators Update. I'm starting to think it might be a region based issue: are you guys using Windows with English (US) as your main language? I've had issues with C# programs in the past where they misbehaved because of the regional decimal separator; is that a possibility? I've tried setting my region settings to US but it seems it didn't help at all.

**Comment by TripleFrequency** [comment]
This definitely could be a regional availability problem, but I don't quite get why. In the U.S. I can download many tracks which normally say "not available in your country," That being said, the Deezer API reports that the track is only available in Canada, Mexico, and the United States. So, I would imagine this is the actual reason for its failure.

**Comment by bascurtiz** [comment]
Holland here. Dutch Win10.

**Comment by svbnet** [comment]
Hey, sorry for being late to the party. The problem is that I foolishly set the default playlist format value to be `{PlaylistName}`:
![image](https://user-images.githubusercontent.com/3015323/33603786-a44b48d2-da18-11e7-88d3-44d7f7e73baf.png)

Where it should be `{ServiceName}` instead. Very sorry for not posting this earlier, and it will be fixed in the next version.

**Comment by rafaeldamasceno** [comment]
I can confirm that the playlist is successfully queued when not using the `{PlaylistName}` token! I had changed the default setting before, but it was to add `\{Title}` to it. 

So @TripleFrequency, it's absolutely not an API related issue. 

Say @svbnet , if I did want the path to be `{PlaylistName}\{Title}`. what would I do then? By `{PlaylistName}` I suppose it refers to the playlist I just added, or is that unsupported by this plugin?
